app/src/main/java/org/piramalswasthya/sakhi/configuration/Dataset.kt

```
@@ -82,6 +83,30 @@ abstract class Dataset(context: Context, val currentLanguage: Languages) {
+        }
+
+        fun dateReverseFormat(dateStr: String): String? {
+            if (dateStr.isEmpty()) return null
+
+            return try {
+                val inputFormat = SimpleDateFormat("dd-MM-yyyy", Locale.getDefault())
+                val outputFormat = SimpleDateFormat("yyyy-MM-dd", Locale.getDefault())
+                
+                val dateResponse = inputFormat.parse(dateStr) ?: return null
+                outputFormat.format(dateResponse)
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)